File: nodejs/src/state/DeliveredState.ts
import OrderState, { IOrder } from "./OrderState";

class DeliveredState extends OrderState {
    constructor(order: IOrder) { super(order); }

    getName(): string { return "DELIVERED"; }

    next(): void {
        console.log(`⚠️ Đơn #${this.order.id}: Đã giao rồi, không thể tiến thêm`);
    }

    cancel(): void {
        console.log(`⚠️ Đơn #${this.order.id}: Đã giao rồi, không thể huỷ`);
    }
}

export default DeliveredState;

File: spring/demo/src/main/java/com/example/demo/state/DeliveredState.java
package com.example.demo.state;

import com.example.demo.model.Order;

public class DeliveredState extends OrderState {
    public DeliveredState(Order order) {
        super(order);
    }

    @Override
    public String getName() {
        return "DELIVERED";
    }

    @Override
    public void next() {
        System.out.println("Đơn #" + order.getId() + ": Đơn đã giao xong");
    }

    @Override
    public void cancel() {
        System.out.println("Đơn #" + order.getId() + ": Không thể hủy khi đã giao");
    }
}
